cnd/src/btsieve/bitcoin/mod.rs

```diff
@@ -84,10 +85,8 @@ where
 
         // Look back into the past (upto timestamp) for one block.
 
-        if let (Some(block), Some(reference_timestamp)) =
-            (oldest_block.as_ref(), reference_timestamp)
-        {
-            if !block.predates(reference_timestamp) {
+        if let Some(block) = oldest_block.as_ref() {
+            if !block.predates(start_of_swap.timestamp() as u32) {
```

Comment:
Instead of doing the conversion here, it would be more ergonomic to change the trait.

Happy for this to be done in a follow-up PR :)

Reply:
If we change the trait we end up needing to cast u32 up to i64, this however saves using the clippy allow attributes.

Reply:
Went with using i64 :)

Reply:
Why no use `NaiveTimestamp` or what the type is called in the trait?

Reply:
Oh, I didn't see that one.